www/js/bluetooth/blueoothScanner.ts

```diff
@@ -0,0 +1,66 @@
+import { logDebug, displayError } from '../plugin/logger';
+
+// Device data, as defined in BluetoothClassicSerial's docs
+type BluetoothClassicDevice = {
+  class: number;
+  id: string;
+  address: string;
+  name: string;
+};
+
+/**
+ * gatherBluetoothData scans for viewable Bluetooth Classic Devices
+ * @param t is the i18next translation function
+ * @returns an array of strings containing device data, formatted ['ID: id Name: name']
+ */
+export default function gatherBluetoothData(t): Promise<string[]> {
+  return new Promise((resolve, reject) => {
+    logDebug('Running bluetooth discovery test!');
+
+    // Device List "I/O"
+    function handleLogs(devices: Array<BluetoothClassicDevice>) {
+      let logs: string[] = [];
+      devices.forEach((device) => {
+        logs.push(
+          `${t('bluetooth.device-info.id')}: ${device.id} ${t('bluetooth.device-info.name')}: ${
+            device.name
+          }`,
+        );
+      });
+      return logs;
+    }
+
+    // Plugin Calls
+    const unpairedDevicesPromise = new Promise((res, rej) => {
+      window['bluetoothClassicSerial'].discoverUnpaired(
+        (devices: Array<BluetoothClassicDevice>) => {
+          res(handleLogs(devices));
```

Comment:
I think we should include, in the logs, whether it is an unpaired or paired device. Since we get two separate callbacks, we should know this. And if we know it, we should log it in this exploratory stage.

```suggestion
          res(handleLogs("unpaired", devices));
```